jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/UriCompliance.java

```diff
@@ -352,6 +352,21 @@ public UriCompliance without(String name, Violation... violations)
         return new UriCompliance(name, remainder);
     }
 
+    /**
+     * Test if violation is referencing a URI path violation.
+     *
+     * @param violation the violation to test.
+     * @return true if violation is a path violation.
+     */
+    public static boolean isPathViolation(UriCompliance.Violation violation)
+    {
+        return (violation == Violation.AMBIGUOUS_PATH_PARAMETER) ||
```

Comment:
This should be an enumset

Reply:
It makes the removal of individual items on a collection WAY more complicated and slower.

Reply:
EnumSet's are highly optimized for enums with less than 32 values.  A contains operation will be a single bitwise operation and null check.  OK in this case, it is probably hardly different from 4 == and 3 || operations, but I think the code is also more readable.  Also we do other EnumSet comparisons on Violations  (e.g. org.eclipse.jetty.http.UriCompliance#AMBIGUOUS_VIOLATIONS) so best to keep the style the same.


```java
    private static final EnumSet<Violation> PATH_VIOLATIONS = Collections.unmodifiableSet(EnumSet.of( 
        Violation.AMBIGUOUS_PATH_SEGMENT,            
        Violation.AMBIGUOUS_PATH_SEPARATOR,
        Violation.AMBIGUOUS_PATH_ENCODING,
        Violation.AMBIGUOUS_EMPTY_SEGMENT));

    // ...
    
    public static boolean isPathViolation(UriCompliance.Violation violation)
    {
        return PATH_VIOLATIONS.contains(violation);
    } 
```

Reply:
Change to EnumSet pushed.